utils/terrascript_client.py

```diff
@@ -306,6 +310,66 @@ def populate_additional_providers(self, accounts):
                            alias=alias,
                            assume_role={'role_arn': assume_role})
 
+    def populate_route53(self, desired_state):
+        for zone in desired_state:
+            # provider = zone['provider']
+            account = zone['account']
+            account_name = account['name']
+            origin = zone['origin']
+            records = zone['records']
+
+            # terraform identifier for zone
+            # We need to replace some characters that are not allowed
+            zone_id = origin
+            zone_id = zone_id.replace(".", "__")
+
+            values = {
+                'name': origin,
+            }
+
+            zone_resource = aws_route53_zone(zone_id, **values)
+            self.add_resource(account_name, zone_resource)
+
+            counts = defaultdict(int)
+            for record in records:
+                counts[record['name']] += 1
+
+                # Terraform record id
+                # This contains a lot of info on purpose
+                # - Needs to be unique per record per zone
+                # - Adding a count for when we have record with the same name
+                # - Terraform client logs only the resource name, so having
+                #   all this info as actually useful when looking at reports
+                record_id = record['name'] + "_" + \
+                    str(counts[record['name']]) + "__" + \
+                    zone_id + "__" + \
+                    record['type'] + "__" + \
+                    record['target'].replace(".", "__")
+
+                # terraform identifier for record
+                # We need to replace some characters that are not allowed
+                record_id = record_id.replace(".", "__")
+                record_id = record_id.replace("@", "_ORIGIN_")
+                record_id = record_id.replace("*", "_wildcard_")
+
+                values = {
+                    'zone_id': zone_resource.fullname,
+                    'name': record['name'],
+                    'type': record['type'],
+                }
+
+                if isinstance(record['target'], str):
+                    values['records'] = [record['target']]
+                else:
+                    msg = f"record {record['name']} for zone {origin} " \
+                          f"is not a string (got: {type(record['target'])}"
+                    logging.error(msg)
+
+                values['ttl'] = record.get('ttl') or zone.get('default_ttl')
```

Comment:
According to rfc2181, TTL `0` is valid, so we should use it instead of assuming the `default_ttl` when that's the case.

Reply:
The idea of `default_ttl` is to let us define a value to use globally on a zone for records that don't have any. This is to simplify and reduce repetition as we typically have the same TTL on all records.
A TTL of zero is almost always not what you want and I've never seen it used ever. It should not be the default.

Reply:
So, if `record.get('ttl')` returns `0`, which is a valid ttl, should we still use the `default_ttl` instead? Just making sure we have the same understanding.